### ISSUE TITLE
unnecessary binding in if-let statement

### ISSUE THREAD

**Issue by Qelxiros** [rust-lang/rust-clippy#13071]
### What it does

Checks for if-let statements that bind a variable where a single value could be specified instead.

### Advantage

Improves readability by replacing a variable with its value and potentially removing a nested if statement.

### Drawbacks

Somewhat niche use case, could be prone to false negatives.

### Example

```rust
let x = Some(("foo", true));

if let Some((message, enabled)) = x {
    if enabled {
        println!("{message}");
    }
}
```

Could be written as:

```rust
let x = Some(("foo", true));

if let Some((message, true)) = x {
    println!("{message}");
}
```
